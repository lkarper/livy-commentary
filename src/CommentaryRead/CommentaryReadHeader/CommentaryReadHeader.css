.CommentaryReadHeader__header.widescreen {
    display: none;
}

.CommentaryReadHeader__header h2 {
    font-size: 16px;
}

.CommentaryReadHeader__a {
    color: black;
    text-decoration: none;
}

.CommentaryReadHeader__a:hover, .CommentaryReadHeader__a:focus {
    text-decoration: underline;
}

@media screen and (min-width: 900px) {
    .CommentaryReadHeader__header.widescreen {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
}