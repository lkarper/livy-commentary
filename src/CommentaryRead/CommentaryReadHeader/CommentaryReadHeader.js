import React from 'react';
import { Link } from 'react-router-dom';
import './CommentaryReadHeader.css';

const CommentaryReadHeader = (props) => {
    const {
        bookNumber,
        chapterNumber,
        sectionNumber,
        suffix,
    } = props;

    const bookLink = (
        <Link 
            to={`/commentary-read/${bookNumber}`}
        >
            Book {bookNumber}
        </Link>
    );

    const chapterLink = (
        <Link
            to={`/commentary-read/${bookNumber}-${chapterNumber}`}
        >
            Chapter {chapterNumber}
        </Link>
    );

    if (bookNumber && chapterNumber && sectionNumber) {
        return (
            <header
                className={`CommentaryReadHeader__header ${suffix}`}
            >
                <h2>{bookLink}{' > '}{chapterLink}{' > '}ยง{sectionNumber}</h2>
            </header>
        );
    } else if (bookNumber && chapterNumber) {
        return (
            <header
                className={`CommentaryReadHeader__header ${suffix}`}
            >
                <h2>{bookLink}{' > '}Chapter {chapterNumber}</h2>
            </header>
        );
    }

    return (
        <header
            className={`CommentaryReadHeader__header ${suffix}`}
        >
            <h2>Book {bookNumber}</h2>
        </header>
    );

}

CommentaryReadHeader.defaultProps = {
    suffix: '',
}

export default CommentaryReadHeader;
