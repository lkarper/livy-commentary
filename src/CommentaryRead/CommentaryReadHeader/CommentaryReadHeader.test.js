import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryReadHeader from './CommentaryReadHeader';

describe('CommentaryReadHeader component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CommentaryReadHeader />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CommentaryReadHeader />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
