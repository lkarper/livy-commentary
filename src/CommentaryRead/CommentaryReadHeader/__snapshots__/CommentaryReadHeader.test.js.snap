// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryReadHeader component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CommentaryReadHeader
      bookNumber=""
      chapterNumber=""
      sectionNumber=""
      suffix=""
    >
      <header
        className="CommentaryReadHeader__header "
      >
        <h2>
          Book 
        </h2>
      </header>
    </CommentaryReadHeader>
  </Router>
</BrowserRouter>
`;
